### PR TITLE
V3 docs

### PR DESCRIPTION
**Pull request by manast**
This PR will keep all the documentation changes for 3.0.0

### CODE REVIEW

**Comment by manast** [comment]
@bradvogel I will keep in this branch all documentation changes for 3.0.0.

**Comment by bradvogel** [comment]
Thanks for fixing up the merge conflicts!

**Comment by bradvogel** [comment]
Anything I can help with to get this merged?

**Comment by manast** [comment]
Basically what is left is to document the new features, such as repeatable jobs, etc. But since I implemented them, it is easier that I also document them. I will try to do it before weeks end.

File: PATTERNS.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+sendQueue.add({msg:"Hello"});
+```
+
+Server B:
+
+```js
+var Queue = require('bull');
+
+var sendQueue = Queue("Server A");
```
Comment (bradvogel) [inline PATTERNS.md]: use `new Queue` here to be consistent

File: PATTERNS.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+var receiveQueue = Queue("Server B");
```
Comment (bradvogel) [inline PATTERNS.md]: use `new Queue` here to be consistent

File: PATTERNS.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+feature in some usecases. For example, you can have two servers that need to
+communicate with each other. By using a queue the servers do not need to be online at the same time, this create a very robust communication channel. You can treat `add` as *send* and `process` as *receive*:
+
+Server A:
+
+```js
+var Queue = require('bull');
+
+var sendQueue = Queue("Server B");
+var receiveQueue = Queue("Server A");
```
Comment (bradvogel) [inline PATTERNS.md]: use `new Queue` here to be consistent

File: PATTERNS.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+that certain keys are placed in the same hash slot, read more about hash tags in the [redis cluster
+tutorial](https://redis.io/topics/cluster-tutorial). 
+
+A hash tag is defined with brakets. I.e. a key that has a substring inside brackets will use that 
+substring to determine in which hash slot the key will be placed. So to make bull compatible with
+cluster, just use a queue prefix inside brackes, for example:
+
+```js
+  var queue = new Queue('cluster', {
+    prefis: '{myprefix}'
```
Comment (bradvogel) [inline PATTERNS.md]: `prefix`?

File: REFERENCE.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+
+
+Queue
+-----
+
+```ts
+Queue(queueName: string, url?: string, opts?: QueueOptions): Queue
+```
+
+The optional ``url``` argument, allows to specify a redis connection string such as for example:
```
Comment (bradvogel) [inline REFERENCE.md]: Mismatch backticks - use three before `url`

**Comment by bradvogel** [comment]
Looks like changes in my PR https://github.com/OptimalBits/bull/pull/507/files didn't make it in here.

**Comment by manast** [comment]
hmm, that sucks. I will add the changes manually or something.

**Comment by manast** [comment]
actually it seems that the changes from PR 507 are indeed there :).

**Comment by bradvogel** [comment]
Are you sure? I don't see the Advanced Settings section in this branch: https://github.com/OptimalBits/bull/pull/507/files#diff-04c6e90faac2675aa89e2176d2eec7d8R432

**Comment by manast** [comment]
weird, so the rest of the things are there but not the part with advanced settings, I don't understand how that could happen. I will add it manually as well.

**Comment by bradvogel** [comment]
Thanks! After that, please merge :)

**Comment by marshall007** [comment]
@manast it appears that in splitting out `PATTERNS.md` into a separate file, the change making `opts.createClient` a top-level property (rather than being nested in `opts.redis.opts.createClient`) was missed.

**Pull request merged**